lib/join/legacyjoin/joinserver.go

```diff
@@ -38,6 +39,17 @@ import (
 	"github.com/gravitational/teleport/lib/tlsca"
 )
 
+// Disabled returns true if the legacy join service is disabled in this environment.
+func Disabled() bool {
+	return os.Getenv("TELEPORT_UNSTABLE_NO_AGENT_ID_SELECTION") != ""
```

Comment:
We've not historically been consistent, but I think for newer flags we've been trying to unify around parsing them with strconv.Parse.
```suggestion
   disabled, _ = strconv.Parse(os.Getenv("TELEPORT_UNSTABLE_NO_AGENT_ID_SELECTION"))
   return disabled
```

Reply:
I can change it then. strconv.ParseBool was actually my first inclination, however, after a quick `rg TELEPORT_UNSTABLE` I saw we are explicitly checking for "yes" on many of these flags, which strconv.Parse doesn't parse as truthy. I didn't want to introduce the one env var that doesn't accept "yes" and some others were already just testing for non-empty as I did here